src/view/widgets/filter_widget/state.rs

```diff
@@ -233,21 +234,30 @@ pub struct FilterListDynamic<T> {
 
 impl FilterListDynamic<AuthorState> {
     fn search_authors(&mut self, tx: UnboundedSender<FilterEvents>) {
-        let name = self.get_name();
-        tokio::spawn(async move {
-            let res = MangadexClient::global().get_authors(&name).await;
-            tx.send(FilterEvents::LoadAuthors(res.ok())).ok();
-        });
+        let name_to_search = SearchTerm::trimmed_lowercased(self.search_bar.value());
+        if let Some(search_term) = name_to_search {
+            tokio::spawn(async move {
+                let response = MangadexClient::global().get_authors(search_term).await;
+                if let Ok(res) = response {
+                    tx.send(FilterEvents::LoadAuthors(res.json().await.ok())).ok();
+                }
+            });
+        }
     }
 }
 
 impl FilterListDynamic<ArtistState> {
     fn search_artists(&mut self, tx: UnboundedSender<FilterEvents>) {
-        let name = self.get_name();
-        tokio::spawn(async move {
-            let res = MangadexClient::global().get_authors(&name).await;
-            tx.send(FilterEvents::LoadArtists(res.ok())).ok();
-        });
+        let name_to_search = SearchTerm::trimmed_lowercased(self.search_bar.value());
+        if let Some(search_term) = name_to_search {
+            tokio::spawn(async move {
+                let response = MangadexClient::global().get_authors(search_term).await;
+
+                if let Ok(res) = response {
+                    tx.send(FilterEvents::LoadArtists(res.json().await.ok())).ok();
+                }
+            });
+        }
```

Comment:
both functions are the exact same but the only thing diferent is the variant of the enum `FilterEvents` sent on success, maybe investigate how to merge both functions into one